Libs/vtkSegmentationCore/Testing/vtkClosedSurfaceToFractionalLabelMapConversionTest1.cxx

```diff
@@ -0,0 +1,137 @@
+
+==============================================================================*/
+
+// VTK includes
+#include <vtkImageAccumulate.h>
+#include <vtkNew.h>
+#include <vtkPolyData.h>
+#include <vtkSphereSource.h>
+
+// vtkSegmentationCore includes
+#include "vtkClosedSurfaceToFractionalLabelmapConversionRule.h"
+#include <vtkOrientedImageData.h>
+#include <vtkPolyDataToFractionalLabelmapFilter.h>
+#include <vtkSegment.h>
+#include <vtkSegmentation.h>
+#include <vtkSegmentationConverter.h>
+#include <vtkSegmentationConverterFactory.h>
+
+void CreateSpherePolyData(vtkPolyData* polyData);
+
+//----------------------------------------------------------------------------
+int vtkClosedSurfaceToFractionalLabelMapConversionTest1(int vtkNotUsed(argc), char* vtkNotUsed(argv)[])
+{
+  // Register converter rules
+  vtkSegmentationConverterFactory::GetInstance()->RegisterConverterRule(
+    vtkSmartPointer<vtkClosedSurfaceToFractionalLabelmapConversionRule>::New() );
+
+  // Generate sphere model
+  vtkNew<vtkPolyData> spherePolyData;
+  CreateSpherePolyData(spherePolyData.GetPointer());
+
+  // Create segment
+  vtkNew<vtkSegment> sphereSegment;
+  sphereSegment->SetName("sphere1");
+  sphereSegment->AddRepresentation(
+    vtkSegmentationConverter::GetSegmentationClosedSurfaceRepresentationName(), spherePolyData.GetPointer());
+
+  // Image geometry used for conversion
+  std::string serializedImageGeometry = "1; 0; 0; 20.7521629333496;"
+                                        "0; 1; 0; 20.7521629333496;"
+                                        "0; 0; 1; 20;"
+                                        "0; 0; 0;  1;"
+                                        "0; 59; 0; 59; 0; 60;";
+
+  // Create segmentation with segment
+  vtkNew<vtkSegmentation> sphereSegmentation;
+  sphereSegmentation->SetConversionParameter(vtkSegmentationConverter::GetReferenceImageGeometryParameterName(), serializedImageGeometry);
+  sphereSegmentation->SetMasterRepresentationName(
+    vtkSegmentationConverter::GetSegmentationClosedSurfaceRepresentationName());
+  sphereSegmentation->AddSegment(sphereSegment.GetPointer());
+
+  sphereSegmentation->CreateRepresentation(vtkSegmentationConverter::GetSegmentationFractionalLabelmapRepresentationName());
+  if (!sphereSegment->GetRepresentation(vtkSegmentationConverter::GetSegmentationFractionalLabelmapRepresentationName()))
+  {
+    std::cerr << __LINE__ << ": Failed to add fractional labelmap representation to segment!" << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  vtkOrientedImageData* fractionalLabelmap = vtkOrientedImageData::SafeDownCast(
+    sphereSegment->GetRepresentation(vtkSegmentationConverter::GetSegmentationFractionalLabelmapRepresentationName()) );
+
+  vtkNew<vtkImageAccumulate> imageAccumulate;
+  imageAccumulate->SetInputData(fractionalLabelmap);
+  imageAccumulate->Update();
+
+  int expectedVoxelCount = 234484;
+  int voxelCount = imageAccumulate->GetVoxelCount();
+  if (voxelCount != expectedVoxelCount)
+  {
```

Comment:
nitpick: Consider using the `CHECK_*` macro. For example see https://github.com/Slicer/Slicer/blob/56e72e1b147834a386f67ade2f71987f0a48a7b3/Modules/Loadable/Volumes/Testing/Cxx/vtkSlicerVolumesLogicTest1.cxx#L81

See http://apidocs.slicer.org/master/vtkAddonTestingMacros_8h.html#define-members

Reply:
I think vtkSegmentationCore does not depend on vtkAddon. It'll be advertised as a separate library (and a copy will be made available in a standalone git repository, independently from Slicer) and if there is enough interest then it may be actually spun out and maintained as a separate library. So, making it depend on vtkAddon might not be preferable.

Reply:
Makes sense..

Whenever you think it makes sens, we could extract the history and create a dedicated repo.

Then we could either add an `UpdateFromUpstream.sh` script or an other external project ...

Reply:
I agree, we shouldn't add dependencies other than regular VTK.

The dedicated repo is in place: https://github.com/PerkLab/PolySeg
I created this for a publication, but it's identical to vtkSegmentationCore (except this one commit). We can make it an external project whenever.